### PR TITLE
Chain cause exception to service exception in translateAndThrow

### LINKED ISSUE #945: BigQueryException discards exception context
If BigQueryException.translateAndThrow is passed an exception that cannot be handled by BaseServiceException it throws a new exception with the message from the passed exception. However this throws away context.

It would be better if the fallback chained the inner exception.

### PR DESCRIPTION
**Pull request by mziccard**
This fixes #945

### CODE REVIEW

**Comment by aozarov** [comment]
LGTM

File: gcloud-java-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryException.java [hunk 1]
```diff
@@ -44,10 +44,15 @@
   private final BigQueryError error;
 
   public BigQueryException(int code, String message) {
```
Comment (aozarov) [inline gcloud-java-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryException.java]: Any reason for leaving this one, and few others in Datastore public?

Comment (mziccard) [inline gcloud-java-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryException.java]: They're all used in the SPI layer

Comment (aozarov) [inline gcloud-java-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryException.java]: I see. Then in that case I change my mind and suggests to make all public and make the classes final.

**Pull request merged**